ludwig/schema/common_fields.py

```diff
@@ -38,12 +38,18 @@ def ResidualField(
     )
 
 
-def NumFCLayersField(default: int = 0, description: str = None, parameter_metadata: ParameterMetadata = None) -> Field:
+def NumFCLayersField(
+    default: int = 0, description: str = None, parameter_metadata: ParameterMetadata = None, non_zero=False
+) -> Field:
     description = description or "Number of stacked fully connected layers to apply."
     full_description = description + (
         " Increasing layers adds capacity to the model, enabling it to learn more complex feature interactions."
     )
     parameter_metadata = parameter_metadata or COMMON_METADATA["num_fc_layers"]
+    if non_zero:
+        return schema_utils.PositiveInteger(
+            default=default, allow_none=False, description=full_description, parameter_metadata=parameter_metadata
+        )
```

Comment:
At the top of the function, add an assertion that if the `default=0` then `non_zero` must be `False`.

Reply:
+1

Reply:
Adding

Reply:
@justinxzhao @ksbrar by the way, this would mean that the test I'm adding will fail with an `AssertionError` and not config validation error